test/socketio.js

```diff
@@ -0,0 +1,129 @@
+"use strict";
+
+const request = require("request-promise");
+const tough = require("tough-cookie");
+const users = require("./data/users");
+const classrooms = require("./data/classrooms");
+
+const io = require("socket.io-client");
+const baseUrl = 'http://localhost:3000';
+const options = {  
+  transports: ['websocket'],
+  'force new connection': true
+};
+
+let authToken;
+const sleep = (t) => new Promise((r) => setTimeout(r, t));
+
+// This series of tests uses user[2] for registration. login, and manipulation
+describe("Socket.IO", () => {
+    let client1, client2, client3;
+
+    it("should connect to the server successfully", (done) => {
+        client1 = io(baseUrl, options);
+        client1.on("connect",()=>{
+            expect(client1.connected).to.equal(true);
+            client1.disconnect();
+            done();
+        });
+        sleep(2000)
+        .then(_=>{client1.disconnect(); done(new Error("Timed out"))});
```

Comment:
calling done twice isn't a problem? I actually don't know if this could cause issues